tests/raftstore/node.rs

```diff
@@ -43,10 +45,26 @@ impl ChannelTransport {
 
 impl Transport for ChannelTransport {
     fn send(&self, msg: raft_serverpb::RaftMessage) -> Result<()> {
+        let from_store = msg.get_from_peer().get_store_id();
         let to_store = msg.get_to_peer().get_store_id();
+        let to_peer_id = msg.get_to_peer().get_id();
+        let region_id = msg.get_region_id();
+        let is_snapshot = msg.get_message().get_msg_type() == raftpb::MessageType::MsgSnapshot;
 
         match self.routers.get(&to_store) {
-            Some(h) => h.rl().send_raft_msg(msg),
+            Some(h) => {
+                try!(h.rl().send_raft_msg(msg));
+                if is_snapshot {
+                    // should report snapshot finish.
+                    self.routers
+                        .get(&from_store)
+                        .unwrap()
+                        .rl()
+                        .report_snapshot(region_id, to_peer_id, SnapshotStatus::Finish)
+                        .unwrap();
```

Comment:
So if send returns ok in `peer.rs` and snapshot does not finish. We will not report Finish nor Failure?